server/service/datapool/validate_data_handler.go

```diff
@@ -67,7 +69,7 @@ func (svc *dataPoolService) handleValidateData(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	filename := svc.Store.MakeRandomFilename()
+	filename := base64.StdEncoding.EncodeToString(dataHash)
```

Comment:
When store the data, the hash of data is used for filename. If we use randomly generated id, we cannot retrieve the data from storage, so I changed it.